File: C++/문제_1920번.cpp
#include <iostream>
#include <algorithm>
using namespace std;

//문제 1920번
int arr[100001];
bool bs(int size, int search);
int main(){
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);
    int n,m, i, num;
    cin >> n;
    for(i=0;i<n;i++){
        cin >> arr[i];
    }
    sort(arr, arr+n);
    cin >> m;
    for(i = 0;i<m;i++){
        cin >> num;
        if(bs(n, num) == 1)
            cout << 1 << "\n";
        else
            cout << 0 << "\n";
    }
    return 0;
}
bool bs(int size, int search){
    int start = 0;
    int end = size - 1;
    int mid = (start + end) / 2;
    while(end - start >= 0){
        if(search > arr[mid]){
            start = mid+1;
        }
        else if(search == arr[mid])
            return 1;
        else
            end = mid - 1;
        mid = (start + end)/2;
    }
    return 0;
}

File: JavaScript/문제_1920번.js
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");
let n = +input[0];
let arr = input[1].split(' ');
let m = +input[2];
let search = input[3].split(' ');
arr = arr.map(o => +o);
arr.sort((o1, o2) => {
  return o1 - o2;
});
let result = "";
search.map((o) => {
  let left = +0;
  let right = n - 1;
  let flag = true;
  while (left <= right) {ㅁ
    let mid = parseInt((left + right) / 2);
    if (arr[mid] > +o)
      right = mid - 1;
    else if (arr[mid] < +o)
      left = mid + 1;
    else {
      flag = false;
      break;
    }
  }
  if (!flag)
    result += '1\n';
  else
    result += '0\n';
  
});
console.log(result);